test/common_checks.py

```diff
@@ -0,0 +1,334 @@
+import pytest
+from typing import Callable, Any
+from test.util import (
+    INFO_LENGTH,
+    info_to_name,
+    info_to_ref,
+    info_to_type,
+    info_to_usermeta,
+)
+
+
+def is_info_object(
+    obj: list[str | int | dict[str, str]],
+    obj_name: str | None = None,
+    obj_type: str | None = None,
+    obj_description: str | None = None,
+    n_items: int | None = None,
+    is_fake_set: bool = False,
+) -> None:
+    """Checks whether a mysterious data structure is a KBase info list.
+
+    If the set was not created by the SetAPI (is_fake_set = True), some of the
+    fields will be filled out differently, so the checks reflect this.
+
+    info list structure:
+    obj_id objid - the numerical id of the object.
+    obj_name name - the name of the object.
+    type_string type - the type of the object.
+    timestamp save_date - the save date of the object.
+    obj_ver ver - the version of the object.
+    username saved_by - the user that saved or copied the object.
+    ws_id wsid - the workspace containing the object.
+    ws_name workspace - the workspace containing the object.
+    string chsum - the md5 checksum of the object.
+    int size - the size of the object in bytes.
+    usermeta meta - arbitrary user-supplied metadata about
+        the object.
+
+    :param obj: the mysterious data structure
+    :type info: list[str|int|dict[str, str]]
+    :param obj_name: expected object name, defaults to None
+    :type obj_name: str | None, optional
+    :param obj_type: expected object type, defaults to None
+    :type obj_type: str | None, optional
+    :param obj_description: expected description in the user metadata, defaults to None
+    :type obj_description: str | None, optional
+    :param n_items: how many items are expected in the user metadata, defaults to None
+    :type n_items: int | None, optional
+    :param is_fake_set: whether or not the current set was created by the SetAPI, defaults to False
+    :type is_fake_set: bool, optional
+    """
+    assert isinstance(obj, list)
+    assert len(obj) == INFO_LENGTH
+    for ix in [0, 4, 6, 9]:
+        assert isinstance(obj[ix], int)
+    for ix in [1, 2, 3, 5, 7, 8]:
+        assert isinstance(obj[ix], str)
+    # usermeta field
+    if obj[10]:
+        assert isinstance(obj[10], dict)
+
+    # some of these can be "falsy" values (0, empty string), so check explicitly against None
+    if obj_name is not None:
+        assert info_to_name(obj) == obj_name
+    if obj_type:
+        assert obj_type in info_to_type(obj)
+    if obj_description is not None and not is_fake_set:
+        assert "description" in info_to_usermeta(obj)
```

Comment:
Is that in every object info created / returned by the SetAPI? It doesn't need to be there for other objects.

Reply:
Description should be present for the *Set objects (even if the description is just `""`) - it won't be present in the items in the set.